### PR TITLE
[sui-execution] First step towards allowing versioning move-vm-types

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

First step towards moving `move-vm-types` to being execution versioned. 

## Test plan 

Existing CI tests.

### CODE REVIEW

File: crates/sui-types/src/execution.rs [hunk 1]
```diff
@@ -2,28 +2,19 @@
 // SPDX-License-Identifier: Apache-2.0
```
Comment (tnowacki) [inline crates/sui-types/src/execution.rs:2]: No suggestions, but maybe we should rename this file? just a thought

**Review by tnowacki — COMMENTED** [review]
Can you update the versioning script? I think you might need to to pass CI

File: sui-execution/latest/sui-adapter/src/execution_mode.rs [hunk 2]
```diff
@@ -201,11 +198,6 @@ impl ExecutionMode for System {
 /// BCS bytes!
 pub struct DevInspect<const SKIP_ALL_CHECKS: bool>;
 
-pub type ExecutionResult = (
-    /*  mutable_reference_outputs */ Vec<(Argument, Vec<u8>, TypeTag)>,
-    /*  return_values */ Vec<(Vec<u8>, TypeTag)>,
-);
-
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply sui-execution/latest/sui-adapter/src/execution_mode.rs:208]: NVM I see that this type was just not versioned

File: sui-execution/src/latest.rs [hunk 3]
```diff
@@ -35,6 +34,7 @@ use sui_verifier_latest::meter::SuiVerifierMeter;
 
 use crate::executor;
 use crate::verifier;
+use sui_adapter_latest::execution_mode;
```
Comment (tnowacki) [inline sui-execution/src/latest.rs:37]: How does this look after the cut?

**Review by tnowacki — COMMENTED** [review]
Can you share the toml after running the script?
sui-execution/Cargo.toml

File: crates/sui-types/src/execution.rs (hunk 1 above)
Comment (tzakian) [reply crates/sui-types/src/execution.rs:2]: I'm up for changing it -- although not sure the best name. Maybe `execution_common.rs`?

**Review by tnowacki — APPROVED** [review]

**Pull request merged**